server/config.go

```diff
@@ -98,21 +143,36 @@ func (s *Service) ReplaceConfigSection(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	section := httprouter.GetParamFromContext(ctx, "section")
 	var res interface{}
-	switch section {
-	case "auth":
-		var authConfig chronograf.AuthConfig
-		if err := json.NewDecoder(r.Body).Decode(&authConfig); err != nil {
-			invalidJSON(w, s.Logger)
-			return
-		}
-		config.Auth = authConfig
-		res = newAuthConfigResponse(*config)
-	default:
-		Error(w, http.StatusBadRequest, fmt.Sprintf("received unknown section %q", section), s.Logger)
+	res = newAuthConfigResponse(*config)
+
+	encodeJSON(w, http.StatusOK, res, s.Logger)
+}
+
+// ReplaceAuthConfig replaces the auth section of the global application configuration
+func (s *Service) ReplaceAuthConfig(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	config, err := s.Store.Config(ctx).Get(ctx)
+	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+
+	if config == nil {
+		Error(w, http.StatusBadRequest, "Configuration object was nil", s.Logger)
+		return
+	}
+
+	var res interface{}
```

Comment:
ditto previous comments about getting rid of `interface{}`

Reply:
:thu